modules/mono/glue/GodotSharp/GodotSharp/Core/Basis.cs

```diff
@@ -799,6 +993,46 @@ internal Basis(real_t xx, real_t yx, real_t zx, real_t xy, real_t yy, real_t zy,
             Row2 = new Vector3(xz, yz, zz);
         }
 
+        /// <summary>
+        /// Constructs a Basis matrix from Euler angles in the specified rotation order. By default, use YXZ order (most common).
+        /// </summary>
+        /// <param name="euler">The Euler angles to use.</param>
+        /// <param name="order">The order to compose the Euler angles.</param>
+        public static Basis FromEuler(Vector3 euler, EulerOrder order = EulerOrder.YXZ)
+        {
+            real_t c, s;
+
+            c = Mathf.Cos(euler.x);
+            s = Mathf.Sin(euler.x);
```

Comment:
In .NET 6 we can use the `System.Math.SinCos`/`System.MathF.SinCos` method to calculate sin and cos together which would be faster. Note that we don't expose these methods in `Godot.Mathf`.

Reply:
This is a good idea to optimize later down the line, but I think it's worth looking over the whole C# codebase for instances like this, and also using it now would hinder cherry-picking to 3.x.

Reply:
I'm not sure it would be a problem for cherry-picking since you have a dedicated PR for 3.x, but yeah optimizations can be done in a future PR.

Reply:
Sorry, I should've clarified, I meant that if we added `Godot.Mathf.SinCos` and used it in more places, then any place we use it becomes slightly harder to cherry-pick.